libs/ui-lib/lib/ocm/components/clusterConfiguration/staticIp/components/FormViewHosts/FormViewHostsFields.tsx

```diff
@@ -1,38 +1,116 @@
+import { getFieldId, PopoverIcon } from '../../../../../../common';
 import HostSummary from '../CollapsedHost';
 import { FormViewHost, StaticProtocolType } from '../../data/dataTypes';
 import { getProtocolVersionLabel, getShownProtocolVersions } from '../../data/protocolVersion';
 import { getEmptyFormViewHost } from '../../data/emptyData';
 import { OcmInputField } from '../../../../ui/OcmFormFields';
+import '../staticIp.css';
+import BondsSelect from './BondsSelect';
+import BondsConfirmationModal from './BondsConfirmationModal';
 
 const getExpandedHostComponent = (protocolType: StaticProtocolType) => {
   const Component: React.FC<HostComponentProps> = ({ fieldName, hostIdx }) => {
+    const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+    const { setFieldValue } = useFormikContext();
+    const [bondPrimaryField] = useField(`${fieldName}.bondPrimaryInterface`);
+    const [bondSecondaryField] = useField(`${fieldName}.bondSecondaryInterface`);
+
+    const [useBond, setUseBond] = useState<boolean>(
+      !!bondPrimaryField.value || !!bondSecondaryField.value,
+    );
+
+    const handleUseBondChange = (checked: boolean) => {
+      if (!checked) {
+        setIsModalOpen(true);
+      } else {
+        setUseBond(true);
+      }
+    };
+
+    const handleModalConfirm = () => {
+      setUseBond(false);
+      setFieldValue(`${fieldName}.bondType`, 'active-backup');
+      setFieldValue(`${fieldName}.bondPrimaryInterface`, '');
+      setFieldValue(`${fieldName}.bondSecondaryInterface`, '');
+      setIsModalOpen(false);
+    };
+
+    const handleModalCancel = () => {
+      setIsModalOpen(false);
+    };
     return (
-      <Grid hasGutter>
-        <OcmInputField
-          name={`${fieldName}.macAddress`}
-          label="MAC Address"
-          isRequired
-          data-testid={`mac-address-${hostIdx}`}
-        />
-        {getShownProtocolVersions(protocolType).map((protocolVersion) => (
-          <FormGroup
-            label={`IP address (${getProtocolVersionLabel(protocolVersion)})`}
-            fieldId={getFieldId(`${fieldName}.ips.${protocolVersion}`, 'input')}
-            key={protocolVersion}
-          >
-            <OcmInputField
-              name={`${fieldName}.ips.${protocolVersion}`}
-              isRequired
-              data-testid={`${protocolVersion}-address-${hostIdx}`}
-            />{' '}
+      <>
+        <Grid hasGutter>
+          <FormGroup>
+            <Checkbox
+              label={
+                <>
+                  {'Use bond'}{' '}
+                  <PopoverIcon
+                    noVerticalAlign
+                    bodyContent="Select this to combine two network interfaces for redundancy and/or increased bandwidth."
+                  />
+                </>
+              }
+              isChecked={useBond}
+              onChange={(_event, value) => handleUseBondChange(value)}
+              id={`use-bond-${hostIdx}`}
+            />
```

Comment:
I think if you use `OcmCheckboxField` instead of the plain PF Checkbox, the `isChecked` and `onChange` properties are going to be controlled automatically, you won't need the `useBond` variable and we won't run a risk of having multiple sources of truth.